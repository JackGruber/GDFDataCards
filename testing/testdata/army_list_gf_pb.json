{
    "gameSystem": "gf",
    "armyId": "oqnnu0gk8q6hyyny",
    "armyIds": [
        "oqnnu0gk8q6hyyny"
    ],
    "armyFaction": "Prime Brothers",
    "armyName": "Prime Brothers",
    "modified": "2023-10-14T13:53:26.045Z",
    "saveVersion": 3,
    "listPoints": 1990,
    "list": {
        "id": "EPp74mi8Yjc_",
        "key": "uPnFHCuiHPmF1Eg6lfOR4",
        "name": "Testing Liste",
        "units": [
            {
                "id": "Z2WiRfh",
                "armyId": "oqnnu0gk8q6hyyny",
                "customName": "Prime Leader alias Master",
                "selectionId": "dyRJr",
                "selectedUpgrades": [
                    {
                        "instanceId": "p7Xkjr6qC",
                        "upgradeId": "BCvkhSe",
                        "optionId": "fwO6jFd"
                    },
                    {
                        "instanceId": "jJfJgn62u",
                        "upgradeId": "MovgoCq",
                        "optionId": "NAzSrGO"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "meWSgTo",
                "armyId": "oqnnu0gk8q6hyyny",
                "selectionId": "GqRu4",
                "selectedUpgrades": [
                    {
                        "instanceId": "5-XPKCT8P",
                        "upgradeId": "vIcBJVF",
                        "optionId": "lCeyzzU"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "PoRVpWT",
                "armyId": "oqnnu0gk8q6hyyny",
                "selectionId": "Mkfui",
                "selectedUpgrades": [],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "81WTJ34",
                "armyId": "oqnnu0gk8q6hyyny",
                "selectionId": "ER4Mp",
                "selectedUpgrades": [
                    {
                        "instanceId": "VoOVC7uUD",
                        "upgradeId": "tbNwc",
                        "optionId": "uNapO"
                    },
                    {
                        "instanceId": "IWgBjg3Nx",
                        "upgradeId": "tbNwc",
                        "optionId": "uNapO"
                    },
                    {
                        "instanceId": "-Fry8j2Pk",
                        "upgradeId": "ar5sR",
                        "optionId": "wyKaPTZ"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "Fz19CEf",
                "armyId": "oqnnu0gk8q6hyyny",
                "selectionId": "kuQC8",
                "selectedUpgrades": [
                    {
                        "instanceId": "JQUgxv-Vx",
                        "upgradeId": "SVNGXyF",
                        "optionId": "PlFiaqs"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "mIz9reK",
                "armyId": "oqnnu0gk8q6hyyny",
                "selectionId": "wsSrv",
                "selectedUpgrades": [],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "VZ7HCmt",
                "armyId": "oqnnu0gk8q6hyyny",
                "selectionId": "XSSh3",
                "selectedUpgrades": [
                    {
                        "instanceId": "uklCjrav9",
                        "upgradeId": "WCoSz98",
                        "optionId": "VV9C8"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "YQGLAAj",
                "armyId": "oqnnu0gk8q6hyyny",
                "selectionId": "RSn5M",
                "selectedUpgrades": [
                    {
                        "instanceId": "QvVP_TyiA",
                        "upgradeId": "uTKvirx",
                        "optionId": "raxBZDD"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            }
        ],
        "points": 1990,
        "pointsLimit": 2000,
        "creationTime": "1697291318315",
        "campaignMode": false,
        "competitive": true,
        "gameSystem": "gf",
        "isCloud": false,
        "modified": "2023-10-14T13:53:26.045Z",
        "selectedUnitId": "RSn5M",
        "unitPreview": null
    },
    "favourite": false,
    "armyVersions": [
        {
            "version": "3.1.0",
            "armyId": "oqnnu0gk8q6hyyny"
        }
    ]
}