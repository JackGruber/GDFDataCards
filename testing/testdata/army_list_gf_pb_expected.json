{
    "armyId": "oqnnu0gk8q6hyyny",
    "gameSystem": "gf",
    "gameSystemId": 2,
    "armyName": "Prime Brothers",
    "listPoints": 2000,
    "listName": "Testing Liste",
    "units": [
        {
            "type": "Prime Master",
            "name": "Prime Leader alias Master",
            "armyId": "oqnnu0gk8q6hyyny",
            "id": "Z2WiRfh",
            "defense": 2,
            "quality": 3,
            "upgrades": [
                "A1",
                "B1",
                "C1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Heavy Rifle",
                    "range": 24,
                    "specialRules": [],
                    "ap": "1"
                },
                {
                    "attacks": 2,
                    "count": 1,
                    "name": "Energy Hammer",
                    "specialRules": [
                        {
                            "key": "blast",
                            "name": "Blast",
                            "type": "ArmyBookRule",
                            "label": "Blast(3)",
                            "modify": false,
                            "rating": "3"
                        }
                    ]
                }
            ],
            "specialRules": [
                {
                    "key": "fearless",
                    "name": "Fearless",
                    "rating": "",
                    "label": "Fearless"
                },
                {
                    "key": "hero",
                    "name": "Hero",
                    "rating": "",
                    "label": "Hero"
                },
                {
                    "key": "tough",
                    "name": "Tough",
                    "rating": "3",
                    "label": "Tough(3)"
                }
            ],
            "equipment": [
                {
                    "name": "Captain",
                    "specialRules": [
                        {
                            "key": "battle-rites",
                            "name": "Battle Rites",
                            "type": "ArmyBookRule",
                            "label": "Battle Rites",
                            "rating": ""
                        }
                    ]
                }
            ]
        },
        {
            "type": "Attack Speeder",
            "name": "Attack Speeder",
            "armyId": "oqnnu0gk8q6hyyny",
            "id": "YQGLAAj",
            "defense": 2,
            "quality": 3,
            "upgrades": [
                "E2"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 2,
                    "count": 1,
                    "name": "Twin Strike Missiles",
                    "range": 18,
                    "specialRules": [
                        {
                            "key": "deadly",
                            "name": "Deadly",
                            "rating": "3",
                            "modify": false,
                            "label": "Deadly(3)"
                        }
                    ],
                    "ap": "3"
                },
                {
                    "attacks": 3,
                    "count": 1,
                    "name": "Laser Talon",
                    "range": 24,
                    "specialRules": [],
                    "ap": "3"
                }
            ],
            "specialRules": [
                {
                    "key": "ambush",
                    "name": "Ambush",
                    "rating": "",
                    "label": "Ambush"
                },
                {
                    "key": "fast",
                    "name": "Fast",
                    "rating": "",
                    "label": "Fast"
                },
                {
                    "key": "fearless",
                    "name": "Fearless",
                    "rating": "",
                    "label": "Fearless"
                },
                {
                    "key": "impact",
                    "name": "Impact",
                    "rating": "6",
                    "label": "Impact(6)"
                },
                {
                    "key": "strider",
                    "name": "Strider",
                    "rating": "",
                    "label": "Strider"
                },
                {
                    "key": "tough",
                    "name": "Tough",
                    "rating": "9",
                    "label": "Tough(9)"
                }
            ]
        },
        {
            "type": "Prime ATV",
            "name": "Prime ATV",
            "armyId": "oqnnu0gk8q6hyyny",
            "id": "VZ7HCmt",
            "defense": 2,
            "quality": 3,
            "upgrades": [
                "N1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 4,
                    "count": 1,
                    "name": "Twin Auto-Rifles",
                    "range": 18,
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Heavy Fusion Rifle",
                    "range": 18,
                    "specialRules": [
                        {
                            "key": "deadly",
                            "name": "Deadly",
                            "type": "ArmyBookRule",
                            "label": "Deadly(6)",
                            "modify": false,
                            "rating": "6"
                        }
                    ],
                    "ap": "4"
                }
            ],
            "specialRules": [
                {
                    "key": "fast",
                    "name": "Fast",
                    "rating": "",
                    "label": "Fast"
                },
                {
                    "key": "fearless",
                    "name": "Fearless",
                    "rating": "",
                    "label": "Fearless"
                },
                {
                    "key": "scout",
                    "name": "Scout",
                    "rating": "",
                    "label": "Scout"
                },
                {
                    "key": "strider",
                    "name": "Strider",
                    "rating": "",
                    "label": "Strider"
                },
                {
                    "key": "tough",
                    "name": "Tough",
                    "rating": "6",
                    "label": "Tough(6)"
                }
            ]
        },
        {
            "type": "Desolator Squad",
            "name": "Desolator Squad",
            "armyId": "oqnnu0gk8q6hyyny",
            "id": "81WTJ34",
            "defense": 2,
            "quality": 3,
            "upgrades": [
                "D1",
                "ftzK4"
            ],
            "size": 5,
            "notes": null,
            "weapons": [
                {
                    "attacks": 2,
                    "count": 5,
                    "name": "CCWs",
                    "specialRules": []
                },
                {
                    "attacks": 2,
                    "count": 2,
                    "name": "HE-Launchers",
                    "range": 30,
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Vengeance HE-Launcher",
                    "range": 30,
                    "specialRules": [
                        {
                            "key": "blast",
                            "name": "Blast",
                            "type": "ArmyBookRule",
                            "label": "Blast(3)",
                            "modify": false,
                            "rating": "3"
                        },
                        {
                            "key": "indirect",
                            "name": "Indirect",
                            "type": "ArmyBookRule",
                            "label": "Indirect",
                            "modify": false,
                            "rating": ""
                        }
                    ]
                },
                {
                    "attacks": 1,
                    "count": 2,
                    "name": "AT-Launcher",
                    "range": 30,
                    "specialRules": [
                        {
                            "key": "lock-on",
                            "name": "Lock-On",
                            "type": "ArmyBookRule",
                            "label": "Lock-On",
                            "modify": false,
                            "rating": ""
                        }
                    ],
                    "ap": "2"
                }
            ],
            "specialRules": [
                {
                    "key": "fearless",
                    "name": "Fearless",
                    "rating": "",
                    "label": "Fearless"
                }
            ]
        },
        {
            "type": "Prime Brothers",
            "name": "Prime Brothers",
            "armyId": "oqnnu0gk8q6hyyny",
            "id": "PoRVpWT",
            "defense": 2,
            "quality": 3,
            "upgrades": [
                "A1",
                "D1",
                "H1"
            ],
            "size": 5,
            "notes": null,
            "weapons": [
                {
                    "attacks": 2,
                    "count": 4,
                    "name": "CCWs",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 4,
                    "name": "Heavy Rifles",
                    "range": 24,
                    "specialRules": [],
                    "ap": "1"
                },
                {
                    "attacks": 2,
                    "count": 1,
                    "name": "Auto-Rifle",
                    "range": 18,
                    "specialRules": []
                },
                {
                    "attacks": 2,
                    "count": 1,
                    "name": "CCW",
                    "specialRules": []
                }
            ],
            "specialRules": [
                {
                    "key": "fearless",
                    "name": "Fearless",
                    "rating": "",
                    "label": "Fearless"
                }
            ]
        },
        {
            "type": "Hover Squad",
            "name": "Hover Squad",
            "armyId": "oqnnu0gk8q6hyyny",
            "id": "mIz9reK",
            "defense": 2,
            "quality": 3,
            "upgrades": [],
            "size": 3,
            "notes": null,
            "weapons": [
                {
                    "attacks": 3,
                    "count": 3,
                    "name": "Suppressor Cannons",
                    "range": 24,
                    "specialRules": [
                        {
                            "key": "indirect",
                            "name": "Indirect",
                            "rating": "",
                            "modify": false,
                            "label": "Indirect"
                        }
                    ],
                    "ap": "2"
                },
                {
                    "attacks": 1,
                    "count": 3,
                    "name": "CCWs",
                    "specialRules": []
                }
            ],
            "specialRules": [
                {
                    "key": "fearless",
                    "name": "Fearless",
                    "rating": "",
                    "label": "Fearless"
                },
                {
                    "key": "relentless",
                    "name": "Relentless",
                    "rating": "",
                    "additional": false,
                    "label": "Relentless"
                },
                {
                    "key": "strider",
                    "name": "Strider",
                    "rating": "",
                    "additional": false,
                    "label": "Strider"
                }
            ]
        },
        {
            "type": "Infiltration Squad",
            "name": "Infiltration Squad",
            "armyId": "oqnnu0gk8q6hyyny",
            "id": "meWSgTo",
            "defense": 3,
            "quality": 4,
            "upgrades": [
                "D1",
                "F1"
            ],
            "size": 5,
            "notes": null,
            "weapons": [
                {
                    "attacks": 2,
                    "count": 5,
                    "name": "CCWs",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 5,
                    "name": "Marksman Carbines",
                    "range": 18,
                    "specialRules": [
                        {
                            "key": "rending",
                            "name": "Rending",
                            "rating": "",
                            "modify": false,
                            "label": "Rending"
                        }
                    ]
                }
            ],
            "specialRules": [
                {
                    "key": "fearless",
                    "name": "Fearless",
                    "rating": "",
                    "label": "Fearless"
                },
                {
                    "key": "scout",
                    "name": "Scout",
                    "rating": "",
                    "label": "Scout"
                },
                {
                    "key": "strider",
                    "name": "Strider",
                    "rating": "",
                    "label": "Strider"
                },
                {
                    "key": "medical-training",
                    "name": "Medical Training",
                    "type": "ArmyBookRule",
                    "label": "Medical Training",
                    "rating": "",
                    "count": 1
                }
            ]
        },
        {
            "type": "Aggro Squad",
            "name": "Aggro Squad",
            "armyId": "oqnnu0gk8q6hyyny",
            "id": "Fz19CEf",
            "defense": 2,
            "quality": 3,
            "upgrades": [
                "M1"
            ],
            "size": 3,
            "notes": null,
            "weapons": [
                {
                    "attacks": 4,
                    "count": 3,
                    "name": "Dual Energy Fists",
                    "specialRules": [],
                    "ap": "4"
                },
                {
                    "attacks": 4,
                    "count": 3,
                    "name": "Twin Fist-Pistols",
                    "range": 12,
                    "specialRules": []
                }
            ],
            "specialRules": [
                {
                    "key": "fearless",
                    "name": "Fearless",
                    "rating": "",
                    "label": "Fearless"
                },
                {
                    "key": "tough",
                    "name": "Tough",
                    "rating": "3",
                    "label": "Tough(3)"
                }
            ]
        }
    ]
}